### PR TITLE
Edit docs for preferential usage of containers

### PR DESCRIPTION
**Pull request by rivera10**

### CODE REVIEW

File: markdown/usage/configuration.md [hunk 1]
```diff
@@ -16,20 +16,18 @@ subtitle: How configure Nextflow to work on your system
 # Introduction
 
 One of the strongest features of Nextflow is that it can run on virtually any computational infrastructure.
-It has built-in support for HPC execution schedulers such as Slurm, SGE, PBS, LSF and more as well as cloud compute infrastructure such as AWS Batch and Google Cloud.
+It has built-in support for HPC execution schedulers such as [Slurm](https://slurm.schedmd.com/quickstart.html), [SGE](https://docs.oracle.com/cd/E19680-01/html/821-1541/ciagcgha.html#scrolltoc), [PBS](https://www.openpbs.org/), [LSF](https://www.ibm.com/support/knowledgecenter/en/SSWRJV_10.1.0/lsf_welcome/lsf_welcome.html) and more as well as cloud compute infrastructure such as [AWS Batch](https://aws.amazon.com/batch/) and [Google Cloud](https://cloud.google.com/).
 
-In order to get nf-core pipelines to run properly on your system, you will need to configure Nextflow so that it knows how best to run your analysis jobs.
+Nextflow also supports container engines such as [Docker](https://www.docker.com/), [Singularity](https://sylabs.io/), [Podman](https://podman.io/), as well as the [Conda](https://docs.conda.io/en/latest/) package management system to deploy the pipelines.
```
Comment (jfy133) [inline markdown/usage/configuration.md:21]:
```suggestion
Nextflow also supports container engines such as [Docker](https://www.docker.com/), [Singularity](https://sylabs.io/), [Podman](https://podman.io/), [Charliecloud](https://hpc.github.io/charliecloud/), [Shifter](https://www.nersc.gov/research-and-development/user-defined-images/), as well as the [Conda](https://docs.conda.io/en/latest/) package management system to deploy the pipelines.
```

File: markdown/usage/configuration.md [hunk 2]
```diff
@@ -41,9 +39,26 @@ There are three main types of pipeline configuration that you can use:
 Each nf-core pipeline comes with a set of "sensible defaults" for the resource requirements of each of the steps in the workflow, found in `config/base.config`.
 These are always loaded and overwritten as needed by subsequent layers of configuration.
 
-In addition to this base config, pipelines come with basic config profiles to enable software packaging methods: `docker`, `singularity`, `podman` and `conda`.
-
-Finally, each pipeline comes with a config profile called `test` and `test_full`. These are used for automated pipeline CI tests and will run the workflow with a minimal / full-size public dataset, respectively.
+In addition to this base config, pipelines have configuration "profiles" that can be enabled with the command line flag `-profile`. Multiple profiles can be specified in a comma-separated list (e.g. `-profile test,docker`). The order of arguments is important! They are loaded in sequence, so later profiles can overwrite earlier profiles. Alternatively, you can create your own configuration profiles and supply these to the pipeline when running.
+
+Examples of available profiles:
```
Comment (jfy133) [inline markdown/usage/configuration.md:44]:
```suggestion
nf-core offers a range of basic profiles for configuration of container engines:
```

File: markdown/usage/configuration.md [hunk 3]
```diff
@@ -41,9 +39,26 @@ There are three main types of pipeline configuration that you can use:
...
+Examples of available profiles:  
+- `docker`
+  - A generic configuration profile to be used with [Docker](http://docker.com/)
+  - Pulls software from DockerHub
+- `singularity`
+  - A generic configuration profile to be used with [Singularity](http://singularity.lbl.gov/)
+  - Pulls software from DockerHub
+- `conda`
+  - A generic configuration profile to be used with [Conda](https://conda.io/docs/)
+  - Pulls most software from [Bioconda](https://bioconda.github.io/)
```
Comment (jfy133) [inline markdown/usage/configuration.md:53]:
```suggestion
- `singularity`
  - A generic configuration profile to be used with [Singularity](http://singularity.lbl.gov/)
  - Pulls software from DockerHub
- `podman`
   - A generic configuration profile to be used with [Podman](https://podman.io/)
- `shifter`
   - A generic configuration profile to be used with [Shifter](https://www.nersc.gov/research-and-development/user-defined-images/)
- `charliecloud`
   - A generic configuration profile to be used with [Charliecloud](https://hpc.github.io/charliecloud/)
- `conda`
  - A generic configuration profile to be used with [Conda](https://conda.io/docs/)
  - Pulls most software from [Bioconda](https://bioconda.github.io/)
```

File: markdown/usage/configuration.md [hunk 4]
```diff
@@ -41,9 +39,26 @@ There are three main types of pipeline configuration that you can use:
...
+> NOTE: Please only use Conda as a last resort, i.e., when it's not possible to run the pipeline with Docker or Singularity.
+- `test`
+  - A profile with a complete configuration for automated testing
+  - Includes links to minimal test data so needs no other parameters
+- `test_full`
+  - Similar to `test` but includes a full-size public dataset to test the pipeline
+
+If `-profile` is not specified, the pipeline will run locally and expect all software to be installed and available on the `PATH`. **This is not recommended**.
```
Comment (jfy133) [inline markdown/usage/configuration.md:62]:
```suggestion

Finally, each pipeline comes with a config profile called `test` and `test_full`. These are used for automated pipeline CI tests and will run the workflow with a minimal / full-size public dataset, respectively. They can also be used for performing test(s) run of nf-core pipeline on your infrastructure, before using your own data.
```

**Review by jfy133 — COMMENTED** [review]
Additional suggestions for completeness

File: markdown/usage/configuration.md [hunk 5]
```diff
@@ -41,10 +39,34 @@ There are three main types of pipeline configuration that you can use:
...
+   - A generic configuration profile to be used with [Charliecloud](https://hpc.github.io/charliecloud/)
+- `conda`
+  - A generic configuration profile to be used with [Conda](https://conda.io/docs/)
+  - Pulls most software from [Bioconda](https://bioconda.github.io/)
+> NOTE: Please only use Conda as a last resort, i.e., when it's not possible to run the pipeline with Docker or Singularity.
+- `test`
+  - A profile with a complete configuration for automated testing
+  - Includes links to minimal test data so needs no other parameters
+- `test_full`
+  - Similar to `test` but includes a full-size public dataset to test the pipeline
```
Comment (ewels) [inline markdown/usage/configuration.md:65]:
```suggestion

> NOTE: Please only use Conda as a last resort, i.e., when it's not possible to run the pipeline with Docker or Singularity.
```

**Review by ewels — APPROVED** [review]

**Comment by ewels** [comment]
## Markdown linting is failing

To keep the code consistent with lots of contributors, we run automated code consistency checks.
To fix this CI test, please run:

* Install `markdownlint-cli`
    * On Mac: `brew install markdownlint-cli`
    * Everything else: [Install `npm`](https://www.npmjs.com/get-npm) then [install `markdownlint-cli`](https://www.npmjs.com/package/markdownlint-cli): `npm install -g markdownlint-cli`
* Try to automatically fix errors in your pipeline: `markdownlint . --config .github/markdownlint.yml --fix`
* Check that there are no remaining problems: `markdownlint . --config .github/markdownlint.yml`

Once you push these changes the test should pass, and you can hide this comment :+1:

We highly recommend setting up Black in your code editor so that this formatting is done automatically on save. Ask about it on Slack for help!

Thanks again for your contribution!

**Comment by rivera10** [comment]
Thanks @ewels and @jfy133. For some reason my editor is not showing these lint issues even though I have a markdownlint. Maybe I have something not set up properly. Anyway, thanks a lot!

**Comment by jfy133** [comment]
Sorry! The last lint issues may have been my bad, I made the suggestion via the GitHub suggestion box where you can't see indent sizes very well 😅

**Pull request merged**